### PR TITLE
Search duplicate put

### PR DESCRIPTION
**Pull request by aberaud**

### CODE REVIEW

File: tests/dhtrunnertester.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ class DhtRunnerTester : public CppUnit::TestFixture {
     CPPUNIT_TEST_SUITE(DhtRunnerTester);
     CPPUNIT_TEST(testConstructors);
     CPPUNIT_TEST(testGetPut);
+    CPPUNIT_TEST(testPutDuplicate);
```
Comment (fsimonfc) [inline tests/dhtrunnertester.h:34]: Missing `CPPUNIT_TEST(testPutOverride);`

File: tests/dhtrunnertester.cpp [hunk 2]
```diff
@@ -90,6 +90,59 @@ DhtRunnerTester::testGetPut() {
...
+
+void
+DhtRunnerTester::testPutOverride() {
+    auto key = dht::InfoHash::get("123");
+    auto val = std::make_shared<dht::Value>("meh");
+    val->id = 42;
+    auto val2 = std::make_shared<dht::Value>("hey");
+    val2->id = 42;
+    CPPUNIT_ASSERT_EQUAL(val->id, val2->id);
+    auto val_data = val->data;
```
Comment (fsimonfc) [inline tests/dhtrunnertester.cpp:126]: `val2->data`

**Review by fsimonfc — CHANGES_REQUESTED** [review]

File: tests/dhtrunnertester.cpp (hunk 2 above)
Comment (aberaud) [reply tests/dhtrunnertester.cpp]: Done

File: tests/dhtrunnertester.h (hunk 1 above)
Comment (aberaud) [reply tests/dhtrunnertester.h:34]: Done

**Review by fsimonfc — APPROVED** [review]

**Pull request merged**